packages/opencensus-web-propagation-tracecontext/src/tracecontext-format.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {HeaderGetter, HeaderSetter, Propagation, randomSpanId, randomTraceId, SpanContext} from '@opencensus/web-core';
+
+/**
+ * @fileoverview Utilities to serialize/deserialize trace context (trace ID,
+ * span ID, trace options and optional trace state) to and from the trace
+ * context `traceparent` and `tracestate` headers.
+ * See https://www.w3.org/TR/trace-context/ for details about the Trace Context
+ * header format.
+ */
+
+/** The default trace options. This defaults to unsampled. */
+const DEFAULT_OPTIONS = 0x0;
+
+/** The traceparent header key. */
+const TRACE_PARENT = 'traceparent';
+/** The tracestate header key. */
+const TRACE_STATE = 'tracestate';
+/** Regular expression that represents a valid traceparent header. */
+const TRACE_PARENT_REGEX = /^[\da-f]{2}-[\da-f]{32}-[\da-f]{16}-[\da-f]{2}$/;
+
+/**
+ * Parses a traceparent header value into a SpanContext object, or null if the
+ * traceparent value is invalid.
+ */
+export function traceParentToSpanContext(traceParent: string): SpanContext|
+    null {
+  const match = traceParent.match(TRACE_PARENT_REGEX);
+  if (!match) return null;
+  const parts = traceParent.split('-');
```

Comment:
Does it make sense to validate the length of parts?

Reply:
Yes, though the lengths are validated implicitly by matching the `TRACE_PARENT_REGEX`